File: modules/buffer-example/android/src/main/java/com/example/BufferExample/BufferExampleModule.kt
package com.example.BufferExample

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

const val PING = "ping"
const val PONG = "pong"

class BufferExampleModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("BufferExample")

    Events("pong")

    Function("ping") {
      return@Function PONG.toByteArray(Charsets.UTF_8)
    }

    AsyncFunction("pingAsync") {
      sendEvent(PONG, mapOf(
        "value" to PONG.toByteArray(Charsets.UTF_8)
      ))
    }
  }
}

File: modules/buffer-example/ios/BufferExampleModule.swift
import ExpoModulesCore

let PING = "ping"
let PONG = "pong"

public class BufferExampleModule: Module {
  public func definition() -> ModuleDefinition {
    Name("BufferExample")

    Events("pong")

    Function("ping") {
      return PONG.data(using: .utf8)
    }

    AsyncFunction("pingAsync") {
      self.sendEvent(PONG, [
        "value": PONG.data(using: .utf8)
      ])
    }
  }
}
